### ISSUE TITLE
Training on multiply/partial datasets

### ISSUE THREAD

**Issue by BINARHO** [NVIDIA/DIGITS#171]
As suggested here:
https://groups.google.com/forum/#!topic/digits-users/r_lIuF-PXxs

It will be nice if at the model setup we could use only some of our dataset of choice, or maybe multily datasets. some 'real world' examples for that feature:
1. Train a model with the first X% of the selected dataset
2. Train a model with random X% of the selected dataset
3. Train a model with X% of dataset number 1 and Y% of dataset number 2

this feature could be very useful as it saves me the need to create multiply datasets that are all subsets of one dataset, and in the other hand, it let me define subsets but still train a model on the entire set (just select 100% of every dataset that is part of the whole set).

a nice UI could be to make the 'Select Dataset' a check list, and if a dataset is check a slide bar will become available to choose the percent of the dataset for the training and a checkbox 'shuffle' to enable scenario (2).

**Comment by lukeyeager** [comment]
Cool idea! However, that's not really feasible with any of the Caffe data layers currently (see list [here](http://caffe.berkeleyvision.org/tutorial/layers.html#data-layers)). There are two ways to accomplish what you want:

1. Edit one [or more] of Caffe's data layers to provide an extra argument in the layer description like `start_key` and/or `end_key`. See the list of current arguments [here](https://github.com/NVIDIA/caffe/blob/v0.12.1/src/caffe/proto/caffe.proto#L445-L477). This would require very little work in DIGITS.

2. Make DIGITS cut out a certain percentage of images from your LMDB and create a new LMDB for each training run. That's slow. It wouldn't be as bad if DIGITS supported the [`ImageData`](http://caffe.berkeleyvision.org/tutorial/layers.html#images) data layer type. That layer is slower than a [`Data`](http://caffe.berkeleyvision.org/tutorial/layers.html#database) layer, but splitting up a textfile would pretty quick and easy.

Neither of those solutions are particularly quick or easy. Sorry, I know your request seems simple. But it actually requires a bit of work.

**Comment by BINARHO** [comment]
2. I don't see this as a bad thing, if as a user I need to create 9 subsets at the Dataset phase, what is the different (performance wise) from letting DIGITS creating each subset at training phase? UX-wise this is a lot easier then re-upload subsets by yourself.
from UI standpoint the user may see all the subsets (that DIGITS created previously) as sub-list at the Dataset section, and at the model phase, if the user chooses percentage that was already sliced before, DIGITS will reuse it.

**Comment by lukeyeager** [comment]
Good point, thanks for the suggestion! I'm working on some other things at the moment, but I like this idea as a future enhancement.

**Comment by BINARHO** [comment]
Thanks, keep the great job 😊
